test/app.listen.js

```diff
@@ -1,14 +1,72 @@
 'use strict'
 
+var net = require('net')
 var express = require('../')
+var http = require('http');
+var assert = require('assert');
 
 describe('app.listen()', function(){
+  function getOpenPort(cb) {
+    var server = net.createServer();
+    server.once('error', function(error) {
+      cb(error, null);
+    });
+    server.listen(0, function(){
+      server.close(cb.bind(null, null, server.address().port));
+    });
+  }
+
+  function makeGetRequest(port, cb){
+    http
+      .get('http://localhost:' + port, function(res){
+        var data = ''
+        res.on('data', function(chunk){
+          data += chunk;
+        });
+
+        res.on('end', function(){
+          cb(null, data);
+        });
+      })
+      .on('error', function(error) {
+        cb(error, null);
+      });
+  }
+
   it('should wrap with an HTTP server', function(done){
     var app = express();
 
-    var server = app.listen(9999, function(){
+    var server = app.listen(0, function(){
       server.close();
       done();
     });
   })
+
+  it('should listened to the requested port', function(done){
+    var expectedResponseBody = 'hello world';
+    var server;
+    var app = express()
+      .get('/', function(req, res){
+        res.send(expectedResponseBody);
+      });
+
+    getOpenPort(function(openPortError, port){
+      if(openPortError !== null) {
+        return done(openPortError);
+      }
+
+      server = app.listen(port, function(){
+        makeGetRequest(port, function(getError, responseBody) {
+          try {
+            assert.strictEqual(getError, null);
+            assert.strictEqual(responseBody, expectedResponseBody)
+          } catch (error) {
```

Comment:
Shouldn't there be a `done()` call at the end? It's asynchronous code.
```suggestion
          try {
            assert.strictEqual(getError, null);
            assert.strictEqual(responseBody, expectedResponseBody)
            done()
          } catch (error) {
```

Reply:
Yes, you are absolutely right